src/module_wrap.cc

```diff
@@ -1143,6 +1146,7 @@ Maybe<URL> ResolveSelf(Environment* env,
     return Nothing<URL>();
   }
 
+  ProcessEmitWarning(env, "The resolve self is experimental.");
```

Comment:
Again the warning should be on the return path (the only return path that isn't Nothing).

Also there is a corresponding path for this in the CJS loader.